src/store/portfolioSummary/index.ts

```diff
@@ -0,0 +1,256 @@
+      {
+        params:
+          {
+            sysparm_fields:
+              'sys_id,clins,portfolio,task_order_number,task_order_status,' +
+              'pop_end_date,pop_start_date',
+            sysparm_query: `portfolio.nameIN` + portfolioSummaryList
+              .map(portfolio => portfolio.name)
+          }
+      }
+    )
+    portfolioSummaryList.forEach(portfolio => {
+      // @ts-ignore
+      portfolio.task_orders = allTaskOrderList
+        // @ts-ignore
+        .filter(taskOrder => taskOrder.portfolio.value === portfolio.sys_id);
+      if (!portfolio.task_orders) {
+        portfolio.task_orders = [];
+      }
+    })
+    return portfolioSummaryList;
+  }
+
+  /**
+   * Constructs a single query that get all the clins records across all the task orders. Parses
+   * the response and sets the clins to the respective task order.
+   */
+  @Action({rawError: true})
+  private async setClinsToPortfolioTaskOrders(portfolioSummaryList: PortfolioSummaryDTO[]) {
+    const clins = portfolioSummaryList.map(portfolio => portfolio.task_orders
+      .map(taskOrder => taskOrder.clins));
+    const allClinList = await api.clinTable.getQuery(
+      {
+        params:
+          {
+            sysparm_fields: 'sys_id,clin_number,clin_status,funds_obligated',
+            sysparm_query: `sys_idIN` + clins
+          }
+      }
+    )
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records =
+          allClinList.filter(clin => (taskOrder.clins.indexOf(<string>clin.sys_id) !== -1));
+      })
+    })
+  }
+
+  /**
+   * Constructs a single query that gets all the cost records across all the clins. Parses
+   * the response and sets the costs to the respective clin.
+   */
+  @Action({rawError: true})
+  private async setCostsToTaskOrderClins(portfolioSummaryList: PortfolioSummaryDTO[]) {
+    const clinNumbers: string[] = [];
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          clinNumbers.push(clinRecord.clin_number);
+        })
+      })
+    });
+    const allCostList = await api.costsTable.getQuery(
+      {
+        params:
+          {
+            sysparm_fields: 'sys_id,clin,task_order_number,is_actual,value',
+            sysparm_query: `clinIN` + clinNumbers
+          }
+      }
+    )
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          clinRecord.cost_records =
+            // @ts-ignore
+            allCostList.filter(cost => cost.clin === clinRecord.clin_number &&
+              cost.task_order_number === taskOrder.task_order_number); // FIXME temp code
+          // allCostList.filter(cost => cost.clin?.value === clinRecord.sys_id);//FIXME correct code
```

Comment:
"temp" or "correct" code here?

Reply:
Usage of combination of "clin_number" and "task_order_number" should be updated to "sys_id" once the task (Jason's team?) on changing clin to a sys_id is complete in "costs" table.